src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> inventory) {
+        inventory.put(transaction.getFruit(),
+                inventory.getOrDefault(transaction.getFruit(), 0) + transaction.getQuantity());
+    }
```

Comment:
The implementation of the handle method should also check if the quantity being supplied is a non-negative number. Supplying a negative quantity does not make sense in the context of a 'Supply' operation and could lead to incorrect inventory adjustments.